### PR TITLE
GCMat plugin

### PR DESCRIPTION
**Pull request by JiaZhou-PU**
# Description

Add GCMat plugin and include one example of GCMat in example folder

# Checklist:

- [x] My code follows the [style guidelines](https://watts.readthedocs.io/en/latest/dev/styleguide.html)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have updated the CHANGELOG.md file (if applicable)
- [x] I have successfully run examples that may be affected by my changes

### CODE REVIEW

**Comment by JiaZhou-PU** [comment]
@paulromano Ready for Review

**Comment by JiaZhou-PU** [comment]
@paulromano @nstauff  Ready for review

File: doc/source/user/plugins.rst [hunk 1]
```diff
@@ -472,3 +472,36 @@ As with other plugins, :class:`~watts.PluginACCERT` is used by::
...
+
+GCMat Plugin
+++++++++++++
+
+The :class:`~watts.PluginGCMat` class enables simulations with Argonne’s global critical materials agent-based model (GCMat) to simulate dynamic economic markets that are composed of agents who have complex decision-making behaviors, and interact with and influence each other, possibly indirectly through market signals. 
+
+The GCMat plugin requires a template input file that can be templated as follows:
+
+
+region    final demand agent	final demand product	reference product	unit	2010	2011	2012	2013	2014	2015	2016	2017	2018	2019	2020	2021	2022	2023	2024	2025	2026	2027	2028	2029	2030
```
Comment (nstauff) [inline doc/source/user/plugins.rst]: Use a code-block to improve formatting?

` .. code-block:: jinja `

File: doc/source/user/plugins.rst [hunk 2]
```diff
@@ -472,3 +472,36 @@ As with other plugins, :class:`~watts.PluginACCERT` is used by::
 
     accert_plugin = watts.PluginACCERT('accert_template')
     accert_result = accert_plugin(params)
+
+GCMat Plugin
+++++++++++++
+
+The :class:`~watts.PluginGCMat` class enables simulations with Argonne’s global critical materials agent-based model (GCMat) to simulate dynamic economic markets that are composed of agents who have complex decision-making behaviors, and interact with and influence each other, possibly indirectly through market signals.
```
Comment (nstauff) [inline doc/source/user/plugins.rst]: break down this sentence. `(GCMat). This code simulates...`

File: examples/1App_GCMat/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# 1App_GCMat
+
+## Purpose
+
+This example provides a demonstration for using WATTS explore supply chain dynamics and uncertainty under nuclear scenarios of Uranium fuel demand growth or shrinkage, supply disruptions.
```
Comment (nstauff) [inline examples/1App_GCMat/README.md]: `This example provides a demonstration for using WATTS to explore supply chain dynamics and uncertainty with GCMat under nuclear scenarios of Uranium fuel demand growth or shrinkage, supply disruptions.`

File: examples/1App_GCMat/watts_exec.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+import watts
+from pathlib import Path
+import numpy as np
+import time
+
+params = watts.Parameters()
+template_name = 'gcmat_template.txt'
+
+final_demand_org = {'2025': 129942.80467537, '2026': 131242.232722123, '2027': 132554.655049345, '2028': 133880.201599838, '2029': 135219.003615836, '2030': 136571.193651995, '2031': 137936.905588515, '2032': 139316.2746444, '2033': 140709.437390844, '2034': 142116.531764752, '2035': 143537.6970824}
```
Comment (nstauff) [inline examples/1App_GCMat/watts_exec.py:32]: add comments to explain what those numbers are - are we selecting 1 set of projections throughout 2035?

File: examples/1App_GCMat/watts_exec.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+us_shares = {'2025': 0.167705168506287, '2026': 0.16583640931287, '2027': 0.164092357127913, '2028': 0.162454545105463, '2029': 0.160909323261296, '2030': 0.159448692009461, '2031': 0.158069206226392, '2032': 0.156774330726817, '2033': 0.155566621066295, '2034': 0.154449705570859, '2035': 0.153426207060785}
+europe_shares = {'2025': 0.121418171095092, '2026': 0.119622503016436, '2027': 0.117931399291808, '2028': 0.116333503569193, '2029': 0.11482130552861, '2030': 0.113390051603732, '2031': 0.112036657179276, '2032': 0.110761596604602, '2033': 0.109563494760443, '2034': 0.108442445566371, '2035': 0.107398402556524}
+row_shares = {'2025': 0.562076660398621, '2026': 0.563641087670695, '2027': 0.56497624358028, '2028': 0.566111951325344, '2029': 0.567069371210094, '2030': 0.567861256386808, '2031': 0.622751279451474, '2032': 0.625765072971703, '2033': 0.622194919609123, '2034': 0.622673325674434, '2035': 0.622981308570158}
+
+us_new_demands = {'2025': 293500, '2026': 292100, '2027': 312300, '2028': 313400, '2029': 377000, '2030': 399100, '2031': 314900, '2032': 361100, '2033': 340200, '2034': 337200, '2035': 336800}
+
+# example from 2025 to 2035
+for i in range(2025, 2036):
```
Comment (nstauff) [inline examples/1App_GCMat/watts_exec.py:45]: explain we are running the GCmat at every year

File: examples/1App_GCMat/watts_exec.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: 2022-2023 UChicago Argonne, LLC
+# SPDX-License-Identifier: MIT
+
+"""
+This example demonstrates how to use WATTS to run an
+GCMAT calculation. This example uses a nuclear
+scenario.
```
Comment (nstauff) [inline examples/1App_GCMat/watts_exec.py]: it is not clear to me what type of simulations we are running. Are we running a set of GCMat scenarios with different sensitivity on input parameters? Or just one reference scenario? Please clarify in comments.

File: examples/1App_GCMat/watts_exec.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+print('results_path',results_path)
+# Define some example parameter variations
+end_years = [2040,2080]
+output_folders = [f"output_{year}" for year in end_years]
+
+# Start timing the simulation
+start = time.perf_counter()
+
+# Loop through the variations and run the simulations
+for end_year, output_folder in zip(end_years, output_folders):
```
Comment (nstauff) [inline examples/1App_GCMat/watts_exec.py]: explain why we need a loop for GCMat calculations

**Review by nstauff — CHANGES_REQUESTED** [review]
Thanks @JiaZhou-PU  - this looks quite good but I have a few comments to request improvements in documentation

**Comment by JiaZhou-PU** [comment]
@nstauff Thanks! Comments are addressed, and ready to merge. @paulromano Thank you for the fixes.

**Review by nstauff — APPROVED** [review]
Thanks @JiaZhou-PU

**Pull request merged**